ONSdigital/eq-author-app
# ✨ EAR 946, 1198, 1199, 1215 - Allow choosing legal basis for Themes

### What is the context of this PR?

Allows picking a legal basis for each theme on the themes settings page per [EAR-946](https://collaborate2.ons.gov.uk/jira/browse/EAR-946).

Per [EAR-1198](https://collaborate2.ons.gov.uk/jira/browse/EAR-1198), removes old legal basis selector from the Design tab of the introduction page.

Per [EAR-1199](https://collaborate2.ons.gov.uk/jira/browse/EAR-1199), removes contact details & legal basis from the Preview tab of the introduction page.

The migration referenced by [EAR-1215](https://collaborate2.ons.gov.uk/jira/browse/EAR-1215) was already implemented in prior Themes work - questionnaires should have a `previewTheme` with `legalBasisCode` set to their old `introduction.legalBasis` on migration. Let me know if there are issues with this and we can chat about what we want the new (if needed) behaviour to be.

Other changes:
- Minor refactoring of `eqId` and `formType` components
- Strip whitespace on the API instead of in each component
- Adds `useUpdateTheme` hook for use in preference to `useMutation`

### How to review

- Check you can still change `surveyId`, `eqId` and `formType`
- Check you can see the legal basis radio buttons for each theme (when enabled) & the design matches the prototype
- Check you can switch legal bases and the changes persist post-refresh

### Screenshot
<img width="763" alt="Screenshot 2021-06-01 at 17 37 34" src="https://user-images.githubusercontent.com/60441612/120359889-17b3ae80-c300-11eb-9405-82ac54fe5228.png">

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
